flake.nix

```diff
@@ -22,11 +23,18 @@
       nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
 
       # Backwards compatibility helper for pre Nix2.6 bundler API
-      program = p: with builtins; with (protect p); "${outPath}/bin/${
-        if p?meta && p.meta?mainProgram then
-          meta.mainProgram
-          else (parseDrvName (unsafeDiscardStringContext p.name)).name
-      }";
+      # TODO: The upstreams of the bundlers using this should be updated to use the new api like nix-bundle was.
+      getExe =
+        x:
+        lib.getExe' x (
+          x.meta.mainProgram or (lib.warn
+            "nix-bundle: Package ${
+              lib.strings.escapeNixIdentifier x.meta.name or x.pname or x.name
+            } does not have the meta.mainProgram attribute. Assuming you want '${lib.getName x}'."
+            lib.getName
+            x
+          )
+        );
```

Comment:
Copied the code from upstream in case it's changed one day  it's 2and improved the message